metagpt/roles/di/data_interpreter.py

```diff
@@ -72,14 +136,49 @@ async def _write_and_exec_code(self, max_retry: int = 3):
                 if ReviewConst.CHANGE_WORDS[0] in review:
                     counter = 0  # redo the task again with help of human suggestions
 
-        return code["code"], result, success
+        return code, result, success
 
-    async def _write_code(self):
-        todo = WriteCodeWithoutTools() if not self.use_tools else WriteCodeWithTools(selected_tools=self.tools)
+    async def _write_code(
+        self,
+        counter,
+        plan_status="",
+        tool_info="",
+    ):
+        todo = WriteAnalysisCode()
```

Comment:
Pass in the necessary parameters when instantiating the action to complete the correct priority setting of the context / config